grid-client/deployer/deployer.go

```diff
@@ -383,6 +390,129 @@ func (d *Deployer) Wait(
 	return deploymentError
 }
 
+// BatchDeploy deploys a batch of deployments
+func (d *Deployer) BatchDeploy(ctx context.Context, deployments map[uint32][]gridtypes.Deployment, deploymentsSolutionProvider map[uint32][]*uint64) (map[uint32][]gridtypes.Deployment, error) {
+	deploymentsSlice := make([]gridtypes.Deployment, 0)
+	nodes := make([]uint32, 0)
+	contractsData := make([]substrate.BatchCreateContractData, 0)
+	nodeClients := make(map[uint32]*client.NodeClient)
+
+	mu := sync.Mutex{}
+
+	group, ctx2 := errgroup.WithContext(ctx)
```

Comment:
ctx2 -> ctx

Reply:
`group.Wait` cancels the context so other functions using ctx will immediately stop